app/sessionStore.js

```diff
@@ -823,29 +823,43 @@ module.exports.runPreMigrations = (data) => {
     delete data.sites
   }
 
-  if (data.lastAppVersion) {
-    try {
-      // with version 0.22.13, any file downloaded (including the update itself) would get
-      // quarantined on macOS (per work done with https://github.com/brave/muon/pull/484)
-      // this functionality was then reverted with https://github.com/brave/muon/pull/570
-      //
-      // To fix the executable, we need to manually un-quarantine the Brave executable so that it works as expected
-      if (process.platform === 'darwin' && compareVersions(data.lastAppVersion, '0.22.13') === 0) {
-        let runningAppPath
-        const appDirectories = app.getPath('exe').split(path.sep)
-        // Remove the `Contents`/`MacOS`/`Brave` parts from path
-        if (appDirectories.length > 3) {
-          runningAppPath = path.sep + path.join(...appDirectories.slice(0, appDirectories.length - 3))
+  if (data.lastAppVersion || data.quarantineNeeded) {
+    // with version 0.22.13, any file downloaded (including the update itself) would get
+    // quarantined on macOS (per work done with https://github.com/brave/muon/pull/484)
+    // this functionality was then reverted with https://github.com/brave/muon/pull/570
+    //
+    // To fix the executable, we need to manually un-quarantine the Brave executable so that it works as expected
+    if (process.platform === 'darwin' && (compareVersions(data.lastAppVersion, '0.22.13') === 0 || data.quarantineNeeded)) {
+      const unQuarantine = (appPath) => {
+        try {
+          execSync(`xattr -d com.apple.quarantine "${appPath}"`)
```

Comment:
this will throw an error if the attribute doesn't exist- but the `catch` below properly catches/logs the error